Sources/Subs.php

```diff
@@ -7040,4 +7040,78 @@ function sentence_list($list)
 	return strtr($format, $replacements);
 }
 
+/**
+ * Truncate an array to a specified length
+ *
+ * @param array $array The array to truncate
+ * @param int $max_length The upperbound on the length
+ * @param int $deep How levels in an multidimensional array should the function take into account.
+ * @return array The truncated array
+ */
+function truncate_array($array, $max_length = 1900, $deep = 3)
+{
+    $array = (array) $array;
+
+    $curr_length = 0;
+    $curr_length = array_length($array);
+
+    if ($curr_length <= $max_length)
+        return $array;
+
+    else
+    {
+        // Truncate each element's value to a reasonable length
+        $param_max = floor($max_length / count($array));
+
+        $currentDeep = $deep - 1;
```

Comment:
It would be cleaner and simpler to just use `$deep--` here and replace all following uses of `$currentDeep` with just `$deep`.

Reply:
Cleaner perhaps but I would like to keep it as two separate values for readability. That and also to prevent unwanted side effects in case there's some weird behaviors on using the same var while in recursive. One should always expect the whole "by-value function cannot modify the value of the variable that was passed to it" thing to hold true but you never know.

Reply:
We can agree to disagree on that, then. 🙂 It doesn't make any substantial difference in this case anyway.

Reply:
$var-- is better to read than $var = $var - 1

Reply:
This isn't a case of $var = $var - 1